### PR TITLE
Refactor Snowflake sampling config [sc-6268]

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

Apply sampling config for general profiling connector.

### 🤓 What?

- Refactor Snowflake profile connector to use new sampling config.

### 🧪 Tested?

Tests passed.

### CODE REVIEW

**Comment by mars-lan** [comment]
Could you also run the crawler manually once to make sure that it's working end-to-end?

File: metaphor/snowflake/profile/extractor.py [hunk 1]
```diff
@@ -63,10 +62,10 @@ async def extract(
         self.max_concurrency = config.max_concurrency
         self.include_views = config.include_views
 
-        assert config.sampling_percentage is None or (
-            0 < config.sampling_percentage <= 100
-        ), f"Invalid sample probability ${config.sampling_percentage}, value must be between 0 and 100"
-        self._sampling_percentage = config.sampling_percentage
+        assert config.sampling is not None and (
```
Comment (mars-lan) [inline metaphor/snowflake/profile/extractor.py:65]: Can move this common check (`> 0 && <= 100`) to `SamplingConfig` and leverage the logic in all connectors?

Comment (elic-eon) [reply metaphor/snowflake/profile/extractor.py:65]: Maybe change it after applying `pydantic` validation?

Comment (mars-lan) [reply metaphor/snowflake/profile/extractor.py:65]: SGTM

**Review by mars-lan — APPROVED** [review]

**Comment by mars-lan** [comment]
Will likely need to merge & fix again after https://github.com/MetaphorData/connectors/pull/139

**Comment by elic-eon** [comment]
> Will likely need to merge & fix again after #139

Looks like it can be merge without conflicts

**Pull request merged**